### PR TITLE
addons: handle add-on names containing hyphens

### PR DESCRIPTION
**Pull request by JakaKokosar**
Before:
Screenshot 2021-10-27 at 08 31 16

After:
Screenshot 2021-10-27 at 08 29 51

### CODE REVIEW

File: orangecanvas/application/addons.py [hunk 1]
```diff
@@ -66,10 +66,10 @@ def normalize_name(name):
...
 
 def prettify_name(name):
-    dash_split = name.split('-')
-    # Orange3-ImageAnalytics => ImageAnalytics
-    if len(dash_split) > 1 and dash_split[0].lower() in ['orange', 'orange3']:
-        name = '-'.join(dash_split[1:])
+    # orange, separator, addon-name
+    orange, _, addon = name.partition('-')
+    # also handle addon names like Survival-Analysis
+    name = addon.replace('-', '') if addon else orange
```
[the remark this answers is not in the document]
Comment (JakaKokosar) [reply orangecanvas/application/addons.py]: > What are valid add-on names? Is orange-Survival-analysis valid?

🤷‍♂️

> I'm thinking whether it'd be safer to replace with space and then get rid of eventual double spaces?

You are right. I updated the regex expression accordingly.
``` python
> prettify_name('Orange3-SurvivalAnalysis')
> "Survival Analysis"
> prettify_name('Orange3-Survival-Analysis')
> "Survival Analysis"
> prettify_name('Orange3-Survival-analysis')
> "Survival analysis"
> prettify_name('Orange3-Survival')
> "Survival"
> prettify_name('Orange3')
> "Orange3"

```

File: orangecanvas/application/addons.py [hunk 2]
```diff
@@ -66,12 +66,12 @@ def normalize_name(name):
 
 
 def prettify_name(name):
-    dash_split = name.split('-')
-    # Orange3-ImageAnalytics => ImageAnalytics
-    if len(dash_split) > 1 and dash_split[0].lower() in ['orange', 'orange3']:
-        name = '-'.join(dash_split[1:])
+    # orange, separator, addon-name
+    orange, _, addon = name.partition('-')
```
Comment (janezd) [inline orangecanvas/application/addons.py:70]: This turns `"Survival-Analysis"` into `Analysis`. Shouldn't the first part be discarded only when it's Orange or Orange3?

**Comment by JakaKokosar** [comment]
@janezd I reverted to the previous solution. Only Orange add-on names are handled now.

File: orangecanvas/application/addons.py [hunk 3]
```diff
@@ -69,9 +69,9 @@ def prettify_name(name):
     dash_split = name.split('-')
     # Orange3-ImageAnalytics => ImageAnalytics
     if len(dash_split) > 1 and dash_split[0].lower() in ['orange', 'orange3']:
-        name = '-'.join(dash_split[1:])
+        name = ' '.join(dash_split[1:])
     # ImageAnalytics => Image Analytics  # while keeping acronyms
-    return re.sub(r"(?<!^)([A-Z][a-z]|(?<=[a-z])[A-Z])", r" \1", name)
+    return re.sub(r"(?<!^)([A-Z][a-z]\s\s+|(?<=[a-z])[A-Z])", r" \1", name)
```
Comment (janezd) [inline orangecanvas/application/addons.py:74]: Although this works, I don't really understand it. Why `\s\s+`? As far as I understand, this now only has to insert spaces into camel case names. Wouldn't a simpler `r"(?<!^)(?<! )([A-Z])"` or even `r"(?<=[a-z])([A-Z])"` work as well?

**Comment by JakaKokosar** [comment]
```python
prettify_name('Orange3-AFooBar')
> A Foo Bar
prettify_name('Orange3-FooBar')
> Foo Bar
prettify_name('Orange3-Foo-Bar')
> Foo Bar
prettify_name('Orange3-Foo-Bar-FOOBAR')
> Foo Bar FOOBAR
prettify_name('Orange3-Foo-bar-foobar')
> Foo bar foobar
prettify_name('Orange3-Foo')
> Foo
prettify_name('Orange3-FOOBar')
> FOO Bar
prettify_name('Orange3-A4FooBar')
> A4Foo Bar
prettify_name('Orange3-4Foo')
> 4Foo
prettify_name('Orange3-Foo3Bar')
> Foo3Bar
prettify_name('Orange3')
> Orange3
```

**Comment by janezd** [comment]
> ```python
> prettify_name('Orange3-AFooBar')
> > A Foo Bar
> prettify_name('Orange3-FooBar')
> > Foo Bar
> prettify_name('Orange3-Foo-Bar')
> > Foo Bar
> prettify_name('Orange3-Foo-Bar-FOOBAR')
> > Foo Bar FOOBAR
> prettify_name('Orange3-Foo-bar-foobar')
> > Foo bar foobar
> prettify_name('Orange3-Foo')
> > Foo
> prettify_name('Orange3-FOOBar')
> > FOO Bar
> prettify_name('Orange3-A4FooBar')
> > A4Foo Bar
> prettify_name('Orange3-4Foo')
> > 4Foo
> prettify_name('Orange3-Foo3Bar')
> > Foo3Bar
> prettify_name('Orange3')
> > Orange3
> ```

May I dare to ask you to add these examples as tests, plus those same examples without 'Orange3' prefix? It would be nice to have them ready for the next time somebody needs to change this function.

**Pull request merged**